src/main/java/org/HdrHistogram/StripedWriterReaderPhaser.java

```diff
@@ -10,93 +10,64 @@
 import java.util.concurrent.locks.ReentrantLock;
 
 /**
- * {@link StripedWriterReaderPhaser} instances provide an asymmetric means for synchronizing the execution of
- * wait-free "writer" critical sections against a "reader phase flip" that needs to make sure no writer critical
- * sections that were active at the beginning of the flip are still active after the flip is done. Multiple writers
- * and multiple readers are supported.
- * <p>
- * While a {@link StripedWriterReaderPhaser} can be useful in multiple scenarios, a specific and common use case is
- * that of safely managing "double buffered" data stream access in which writers can proceed without being
- * blocked, while readers gain access to stable and unchanging buffer samples
- * <blockquote>
- * NOTE: {@link StripedWriterReaderPhaser} writers are wait-free on architectures that support wait-free atomic
- * increment operations. They remain lock-free (but not wait-free) on architectures that do not support
- * wait-free atomic increment operations.
- * </blockquote>
- * {@link StripedWriterReaderPhaser} "writers" are wait free, "readers" block for other "readers", and
- * "readers" are only blocked by "writers" whose critical section was entered before the reader's
- * {@link StripedWriterReaderPhaser#flipPhase()} attempt.
- * <p>
- * When used to protect an actively recording data structure, the assumptions on how readers and writers act are:
- * <ol>
- * <li>There are two sets of data structures ("active" and "inactive")</li>
- * <li>Writing is done to the perceived active version (as perceived by the writer), and only
- *     within critical sections delineated by {@link StripedWriterReaderPhaser#writerCriticalSectionEnter}
- *     and {@link StripedWriterReaderPhaser#writerCriticalSectionExit}).</li>
- * <li>Only readers switch the perceived roles of the active and inactive data structures.
- *     They do so only while under readerLock(), and only before calling flipPhase().</li>
- * </ol>
- * When the above assumptions are met, {@link StripedWriterReaderPhaser} guarantees that the inactive data structures are not
- * being modified by any writers while being read while under readerLock() protection after a flipPhase()
- * operation.
- *
- *
+ * {@link StripedWriterReaderPhaser} is a striped version of {@link WriterReaderPhaser}
+ * which reduces contention at the expense of higher space consumption.
  *
  */
-public class StripedWriterReaderPhaser {
+public class StripedWriterReaderPhaser extends WriterReaderPhaser {
+    private static final int PADDING_BYTES = 128;
```

Comment:
Consider changing STRIDE_BYTES to 256. This will make STRIDE 8, and allow the optimizer to reduce the i * STRIDE math in the hit paths to a single shift.

Reply:
Actually this will make STRIDE 32. Currently it's 16 and in both cases it's a power of 2.